lib/pages/profile_page/community_list.dart

```diff
@@ -52,12 +66,38 @@ class CommunityListWidget extends StatelessWidget {
                             fontWeight: FontWeight.w800,
                           ),
                         )),
-                    if (user.username == user.communities![index][1]['creator'])
-                      Icon(
-                        Icons.admin_panel_settings,
-                        color: AppThemes.iconColor(context),
-                        size: 30.0,
-                      ),
+                    Row(
+                      mainAxisAlignment: MainAxisAlignment.end,
+                      children: [
+                        if (computeStreak(
```

Comment:
1. Instead of calling `computeStreak()` twice, call it once, store the value in a variable and use that variable.
2. Are we considering that `first_streak_date` and `last_streak_date` might be null?

Reply:
2. First and last streak will never be null.

Reply:
1. Is there a clean way to do it besides computing the streak for all functions by using a list?